docs/proposals/failover/cluster-failover.md

```diff
@@ -0,0 +1,161 @@
+  verify the new added fields.
+
+## Non-Goals
+
+## Proposal
+
+### Dependency
+
+The failover feature is controlled by the `Failover` feature gate. `Failover` feature gate has evolved to Beta since Karmada v1.4 and is enabled by default.
+When configuring failover behavior in the policy, make sure that `Failover` feature gate is enabled.
+
+The graceful eviction feature is controlled by the `GracefulEviction` feature gate. `GracefulEviction` feature gate has evolved to Beta since Karmada v1.4 and is enabled by default. When configuring failover behavior with purgeMode `Graciously` and `Never`, make sure that `GracefulEviction` feature gate is enabled.
+
+### User Stories (Optional)
+
+#### As a user, I deploy different applications in multiple clusters. I hope they have different `GracePeriodSeconds` for graceful eviction.
+
+Supposing I deploy different applications in multiple clusters. I hope they will be gracefully evicted when detecting a cluster failure. But their image sizes are different, and they may have different startup time on their first initialization. Therefore, I hope they have different `GracePeriodSeconds` so that old replicas will not be deleted before the new replicas start.
+
+## Design Details
+
+![cluster-failover](statics/cluster-failover.png)
+
+### API change
+
+The configuration of failover behavior includes:
+
+* `PurgeMode` which represents how to deal with the legacy applications on the cluster from which the application is migrated.
+* `GracePeriodSeconds` which is the maximum waiting duration in seconds before application on the migrated cluster should be deleted.
+
+#### PropagationPolicy API change
+
+```golang
+// FailoverBehavior indicates failover behaviors in case of an application or
+// cluster failure.
+type FailoverBehavior struct {
+    // Application indicates failover behaviors in case of application failure.
+    // If this value is nil, failover is disabled.
+    // If set, the PropagateDeps should be true so that the dependencies could
+    // be migrated along with the application.
+    // +optional
+    Application *ApplicationFailoverBehavior `json:"application,omitempty"`
+  
+    // Cluster indicates failover behaviors in case of cluster failure.
+    // If not set, Karmada will gracefully evict the workloads.
+    // The default `GracePeriodSeconds` is from `--graceful-eviction-timeout` option 
+    // in karmada-controller-manager.
+    // +optional
+    Cluster *ClusterFailoverBehavior `json:"cluster,omitempty"`
+}
+
+// ClusterFailoverBehavior indicates cluster failover behaviors.
+type ClusterFailoverBehavior struct {
+    // PurgeMode represents how to deal with the legacy applications on the
+    // cluster from which the application is migrated.
+    // Valid options are "Immediately", "Graciously" and "Never".
+    // Defaults to "Graciously".
+    // +kubebuilder:default=Graciously
```

Comment:
```
// +kubebuilder:validation:Enum=Immediately;Graciously;Never
```

Reply:
I think `PurgeMode PurgeMode `json:"purgeMode,omitempty"` has ensured it. We may not need it.